### PR TITLE
feat: Use new test framework

### PR DESCRIPTION
**Pull request by bvestli**
This will use blueprint-test as a new test framework. We also removes the unit tests as these
are not useful for IaC. The one we need is the integration test.

### CODE REVIEW

File: test/integration/cloud_storage_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+package cloud_storage
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/random"
+
+	// import the blueprints test framework modules for testing and assertions
```
Comment (AlexanderBrevig) [inline test/integration/cloud_storage_test.go:11]: Please delete all comments except the one that I've commented (#L16). 
These provide no additional information and your naming and the API makes it clear what's going on 👍🏻

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]
Looking good. Needs a comment cleanup but seems to be a good minimal test case 👍🏻

File: .github/workflows/pr-int-test-terraform.yml [hunk 2]
```diff
@@ -43,8 +43,8 @@ jobs:
           cluster_name: ${{ secrets.CLUSTER_NAME }}
           location: ${{ secrets.CLUSTER_LOCATION }}
       - name: Download Go Modules
-        working-directory: tests
+        working-directory: test/integration
```
[the remark this answers is not in the document]
Comment (bvestli) [reply .github/workflows/pr-int-test-terraform.yml:46]: I think unit and integration tests are a bit unclear when dealing with IaC. In the blueprint test framework they call it integration tests. In the book "Infrastructure as Code, Patterns and Practices" they have the following definition of an integration test: "Integration tests run against testing environments and dynamically analyze infrastructure resources to verify that they are affected by module or configuration changes.". And unit tests as: "Unit tests statically analyze plaintext infrastructure configuration or state. They do not rely on live infrastructure resources or dependencies."
Ref. https://livebook.manning.com/book/infrastructure-as-code-patterns-and-practices/chapter-6/134

So using those definitions, I think it is right to call it integration tests.

[the remark this answers is not in the document]
Comment (AlexanderBrevig) [reply .github/workflows/pr-int-test-terraform.yml:46]: Though `run against` and `affected` indicates that they are in fact surviving the test. We are effectively creating a new environment/configuration. This does not prove that the potential integration of version next does not break configurations of version current.

If this test were performed (and verified no deletions f.ex) without destroy, i.e. not ephemeral, then I would agree. I think the book agrees with this idea.

One way to think about it that kind of makes sense is that we have tested that our module "integrates" with the terraform providers as expected, though in my mind we already use a contract that we should trust so to me this feels like a stretch.
If I'm the only one, I can think of testing against a live configuration more akin to an end to end test and probably still get my sleep ;) .

**Review by AlexanderBrevig — APPROVED** [review]

**Pull request merged**